src/Hydra/Serializer/PartialCollectionViewNormalizer.php

```diff
@@ -168,15 +168,15 @@ private function cursorPaginationFields(array $fields, int $direction, $object):
         return $paginationFilters;
     }
 
-    private function populateDataWithCursorBasedPagination(array $data, array $parsed, \Traversable $object, ?array $cursorPaginationAttribute, ?int $urlGenerationStrategy): array
+    private function populateDataWithCursorBasedPagination(array $data, array $parsed, \Traversable $object, ?array $cursorPaginationAttribute, ?int $urlGenerationStrategy, ?float $itemsPerPage, ?float $pageTotalItems): array
     {
         $objects = iterator_to_array($object);
         $firstObject = current($objects);
         $lastObject = end($objects);
 
         $data['hydra:view']['@id'] = IriHelper::createIri($parsed['parts'], $parsed['parameters'], urlGenerationStrategy: $urlGenerationStrategy);
 
-        if (false !== $lastObject && \is_array($cursorPaginationAttribute)) {
+        if (false !== $lastObject && \is_array($cursorPaginationAttribute) && $pageTotalItems >= $itemsPerPage) {
```

Comment:
when `$pageTotalItems == $itemsPerPage` we want to show hydra:view? Can you test this? I thought that with a cursor we did not know the amount of total items (no count)?

Reply:
We did test it out, we have it live in our production environment. This isn't perfect because it still could return no results on the next page, however, it prevents an empty page from showing hydra:next.

The logic is that with cursor pagination if you no longer have results on a page, the next one won't either because you are doing a less less than or greater than some identifier. Because it's only a comparison like that if it doesn't return enough results we can be certain the next page will return none.

Reply:
would you be able to add a test for this?

Reply:
Sure.

Reply:
Sorry I haven't gotten around to adding a test here. It isn't forgotten though.